compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8910,6 +8910,13 @@ private STNode parseRemoteCallOrAsyncSendEnd(STNode expression, STNode rightArro
             case OPEN_PAREN_TOKEN:
                 return parseRemoteMethodCallAction(expression, rightArrow, name);
             case SEMICOLON_TOKEN:
+            case CLOSE_PAREN_TOKEN:
+            case COMMA_TOKEN:
+            case LET_KEYWORD:
+            case WHERE_KEYWORD:
+            case ORDER_KEYWORD:
+            case LIMIT_KEYWORD:
+            case SELECT_KEYWORD:
```

Comment:
Do we need to add all the intermediate clause start tokens here? `from` and `join` seem to be missing.

We also need to add open-brace as we can have the following. Seems it is a bug. `match action-or-expr { match-clause+ }`